rsp_vision/save/save_data.py

```diff
@@ -0,0 +1,378 @@
+                ]
+                assert len(this_line_rows) >= 1
+                assert len(this_line_rows["sub"].unique()) == 1
+                sub = this_line_rows["sub"].unique()[0]
+
+                ses = (
+                    analysis_log[
+                        (
+                            analysis_log["mouse line"]
+                            == folder_naming_specs.mouse_line
+                        )
+                        & (
+                            analysis_log["mouse id"]
+                            == folder_naming_specs.mouse_id
+                        )
+                    ]["ses"].max()
+                    + 1
+                )
+            else:
+                # CASE 4: this session was analysed before
+                this_line_rows = analysis_log[
+                    (
+                        analysis_log["mouse line"]
+                        == folder_naming_specs.mouse_line
+                    )
+                    & (
+                        analysis_log["mouse id"]
+                        == folder_naming_specs.mouse_id
+                    )
+                ]
+                assert len(this_line_rows) >= 1
+                assert len(this_line_rows["sub"].unique()) == 1
+                sub = this_line_rows["sub"].unique()[0]
+
+                ses = this_line_rows["ses"].max()
+                reanalysis = True
+
+    except FileNotFoundError:
+        #  CASE 0: no analysis log file
+        sub = 0
+        ses = 0
+        analysis_log = pd.DataFrame(
+            columns=[
+                "folder name",
+                "sub",
+                "ses",
+                "mouse line",
+                "mouse id",
+                "hemisphere",
+                "brain region",
+                "monitor position",
+                "fov",
+                "cre",
+                "analysed",
+                "analysis date",
+                "commit hash",
+                "microscope",
+                "n roi",
+                "n responsive roi",
+                "days of the experiment",
+            ],
+        )
+    assert sub is not None
+    assert ses is not None
+
+    return sub, ses, reanalysis, analysis_log
+
+
+def save_roi_info_and_fit_outputs(
+    photon_data: PhotonData,
+    session_folder: SessionFolder,
+):
+    """
+    Save the roi info and the fit outputs in a pickle object.
+
+    Parameters
+    ----------
+    photon_data : PhotonData
+        The PhotonData object.
+    session_folder : SessionFolder
+        The SessionFolder object.
+    """
+    # dict with subset of the data
+    subset = {
+        "n_roi": photon_data.n_roi,
+        "responsive_rois": photon_data.responsive_rois,
+        "median_subtracted_responses": photon_data.median_subtracted_response,
+        "fit_outputs": photon_data.fit_output,
+    }
+
+    #  save part of the data in a pickle object
+    with open(
+        session_folder.ses_folder_path / "gaussians_fits_and_roi_info.pickle",
```

Comment:
Minor issue: the saved file names do not compy with [SWC-Blueprint file naming conventions](https://swc-blueprint.neuroinformatics.dev/specification.html#file-naming-conventions). If it's difficult to make the names compliant don't worry about it, these are conventions, not hard requirements.

Reply:
I feel more comfortable for now with these descriptive names rather than the codes... However, I recognize that in this way we can identify the subject and the session this data belongs to only from the parent folders... which is not ideal...

Reply:
I decided to not change it